study.md

```diff
@@ -46,15 +51,46 @@ other named Mr.MagicNibbles (yellow) and bites a bit. Please copy the JSON you
 already wrote and include the information about Dave's fish.
 
 ```json
-<!-- your answer here -->
+{ "name":"Jason",
+  "things" : {
+    "cat": "Mr. Kitty",
+    "roommates": [
+      { "name": "Dave",
+        "things": {
+          "goldfish": [
+            { "name": "Bob", "color": "red", "bites": false},
+            { "name": "Mr. MagicNibbles", "color": "yellow", "bites": true}
+          ]
+        }
+      },
+      { "name": "Miller", "things": []}
+    ],
+    "loves": ["Red Sox", "Patriots", "Bruins"],
+    "recent-addresses": ["123 Fake Street", "2 Muchinfo Road"]
+  }
+}
 ```
 
 ## JSON Methods
 
 Using the JSON your wrote above, can you write a method on the JSON?
 
-```json
-<!-- your answer here -->
+```json`
+const petRoommatesGoldfish = function(json,roommateName,fishName) {
+  let roomies = json.things.roommates;
+  for ( let i = 0; i < roomies.length; i++ ) {
+    if (roomies[i].name === roommateName && roomies[i].things) {
+      let fishies = (roomies[i].things.goldfish || []);
+      for (let j = 0; j < fishies.length; j++) {
+        if (fishies[j].name === fishName) {
+          return (fishies[j].bites ? "YOW!!!" : "Aww, nice fishy.");
+        }
+      }
+      return `${roomies[i].name} doesn't have a fish by that name.`;
+    }
+  }
+  return "That roommate is imaginary.";
+};
```

Comment:
nice.  I see what you did.  I was looking for a "no, you cant write methods in JSON" but this is golden.  you would have to turn the JSON into an object however by doing a JSON.parse(jason)